niftilib/nifti1_tool.c

```diff
@@ -3885,7 +3885,9 @@ int disp_raw_data( void * data, int type, int nvals, char space, int newline )
          {
                float temp;
                memcpy(&temp, dp, sizeof(temp));
-               snprintf(fbuf,sizeof(fbuf),"%f", temp);
+               snprintf(fbuf, NT_LOC_MAX_FLOAT_BUF,"%f", temp);
+               /* terminate, since some versions might not */
```

Comment:
I've never heard of any `snprintf` with such a disastrous bug...

Are you perhaps thinking of the old Microsoft `_snprintf` (note the leading underscore)?  See:

https://stackoverflow.com/questions/7706936/is-snprintf-always-null-terminating

As we are not calling `_snprintf` this change should not be needed.

Reply:
"Bug" is not for us to decide regarding this, the question is what does the doc say vs what does the function do.  Reading the man page many times, it is not 100% clear that it always does the termination, the help seems a bit ambiguous.  But yes, I guess it was the _ version that raised the flag.
So I don't really care about this either way.  Being safe is good, not mentioning M!cr*s@ft is very good.

Reply:
>"Bug" is not for us to decide regarding this

The C standard requires snprintf to ***always*** null terminate. I've updated sprintf to snprintf in many open source libraries recently, and have never seen the MS version having any problem.  It was only the old non-standard underscore version that was busted.

I would remove this change, it an unnecessary complication.

Reply:
Okay, this is the last question remaining for this PR.  You can choose: keep the termination or remove it, what do you prefer?  I am happy keeping it.  If I have to do google searches to verify the functionality, it is preferable not to make "the next programmer" have to as well, so clarity is nice.

Reply:
Okay, I will remove this, thanks.

Reply:
BTW, I did look at the C standard, and indeed it is quite clear about the null termination, while the man page is not quite so clear.  That should by my go-to reference, rather than the man pages.  Thanks.